docs/advanced.rst

```diff
@@ -361,6 +361,55 @@ And you don't want to store the changes for the field ``pub_date``, it is necess
 
 By default, django-simple-history stores the changes for all fields in the model.
 
+Adding additional fields to historical models
+---------------------------------------------
+
+Sometimes it is useful to be able to add additional fields to historical models that do not exist on the
+source model. This is possible by combining the ``bases`` functionality with the ``pre_create_historical_record`` signal.
+
+.. code-block:: python
+
+    # in models.py
+    class IPAddressHistoricalModel(models.Model):
+        """
+        Abstract model for history models tracking the IP address.
+        """
+        ip_address = models.GenericIPAddressField(_('IP address'))
+
+        class Meta:
+            abstract = True
+
+
+    class PollWithExtraFields(models.Model):
+        question = models.CharField(max_length=200)
+        pub_date = models.DateTimeField('date published')
+
+        history = HistoricalRecords(bases=[IPAddressHistoricalModel,]
+
+
+.. code-block:: python
+
+    # define your signal handler/callback anywhere outside of models.py
+    def add_history_ip_address(sender, **kwargs):
+        instance = kwargs['instance']
+        history_instance = kwargs['history_instance']
+        history_instance.ip_address = instance.request.META['REMOTE_ADDR']
```

Comment:
Let's test this. Don't think the instance has a request. But the thread should have one